server/protocols/protocols-smtp-dkim/src/main/java/org/apache/james/smtpserver/DKIMHook.java

```diff
@@ -165,7 +205,9 @@ public Config(boolean forceCRLF, boolean signatureRequired, Optional<Domain> onl
 
         DKIMCheckNeeded dkimCheckNeeded() {
             return onlyForSenderDomain
-                .map(DKIMCheckNeeded::onlyForSenderDomain)
+                .map(domain -> DKIMCheckNeeded.or(
+                    DKIMCheckNeeded.onlyForSenderDomain(domain),
+                    DKIMCheckNeeded.onlyForHeaderFromDomain(domain)))
```

Comment:
What would be perfect would be to have it configurable.

`<validatedEntities>envelope,headers</validatedEntities>`

Defaults to both if unspecified, coma separated values.

This would enable greater flexibility when configuring this and would allow for easy rollback to previous state.

WDYT?

Reply:
Doable yes, more flexibility is good. Will give it a shot, thanks for the suggestion!

Reply:
Complexifies a bit the code but ok I think.

However I ain't sure how to test it in DKIMHookIntegrationTest. Would need a different smtpserver.xml file? Thought maybe doing like in CertificateReloadTest with the temporary server config reload... but makes the test structure more complex, maybe there is a better way? But maybe not. Wasn't sure ! But I might go with it later